File: pun_of_the_day.py
import requests
from bs4 import BeautifulSoup


class PunOfTheDay:
    def __init__(self):
        pass

    def get_random_pun(self):
        res = requests.get("https://pun.me/random/")
        soup = BeautifulSoup(res.text, 'html.parser')

        all_puns = soup.find_all("ul", {"class": "puns single"})
        all_puns[0].li.span.decompose()
        pun = all_puns[0].li.get_text()

        return pun

File: punOfTheDay.js
const cheerio = require("cheerio");

exports.getRandomPun = async function(fetcher) {
  fetchFunction = fetcher;
  if (!fetchFunction) {
    fetchFunction = fetch;
  }
  const res = await fetchFunction("https://pun.me/random/");
  const text = await res.text();
  const $ = cheerio.load(text);

  $('ul[class="puns single"] li span').remove();

  return $('ul[class="puns single"] li').text();
};
